DependencyInjection/DoctrineExtension.php

```diff
@@ -304,10 +313,22 @@ protected function getConnectionOptions($connection)
                 $options['global'][$key] = $value;
                 unset($options[$key]);
             }
+            if (!isset($options['global']['driver'])) {
+                $options['global']['driver'] = $options['driver'];
+            }
+            foreach ($options['shards'] as $i => $shard) {
+                if (!isset($shard['driver'])) {
+                    $options['shards'][$i]['driver'] = $options['driver'];
```

Comment:
well, then ``doctrine:database:create`` should be fixed instead when it builds params in a special way instead of using the configured params.
Adding useless params is the wrong way (the DriverManager will never read ``$params['shard'][$i]['driver']`` but always ``$params['driver']``)